### PR TITLE
Refactor/docs

### PR DESCRIPTION
**Pull request by SimonCW**
Hi @maciejkula , 

I took a stab at clarifying the documentation wr.t. to the usage of features a bit. I think the docs for the dataset class could also be improved in this regard. What do you think? 

I hope some clarification in the docs could be a first step to avoid issues like these: https://github.com/lyst/lightfm/issues/571, https://github.com/lyst/lightfm/issues/577

I made this a draft because I would like to make the explanations a bit more concise and rework the dataset docs a bit.

### CODE REVIEW

**Comment by maciejkula** [comment]
This looks good to me so far, thanks for taking this on!

**Comment by SimonCW** [comment]
Hi @maciejkula, 
I re-read the changes and added two links in the dataset class. I think this might help some users understand the identity matrix. If you feel the same feel free to merge.

**Comment by SimonCW** [comment]
Mh, I was able to resolve the issue causing unit tests to fail (invalid argument combination to sklearn KFold). I couldn't find out what is going wrong with the MacOS Travis build. 

Can you help @maciejkula?

File: tests/test_movielens.py [hunk 1]
```diff
@@ -704,7 +704,7 @@ def split(self, X, y=None, groups=None):
             for _ in range(self.n_splits):
                 yield idx, idx
 
-    cv = CV(n_splits=3, random_state=42)
+    cv = CV(n_splits=3, random_state=None)
```
Comment (maciejkula) [inline tests/test_movielens.py:707]: Would you mind pointing me at the error this caused?

We shouldn't merge this change - it would make tests non-deterministic.

**Comment by maciejkula** [comment]
The OSX build seems to have been failing due to [this sklearn issue](https://github.com/scikit-learn/scikit-learn/issues/19063). I re-ran it and it seems to have pulled in a new, fixed sklearn version.

File: doc/conf.py [hunk 2]
```diff
@@ -145,7 +145,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+html_static_path = []
```
Comment (maciejkula) [inline doc/conf.py:148]: What does this do?

Comment (maciejkula) [inline doc/conf.py:148]: I'd still like to know why this change is needed :)

**Review by maciejkula — APPROVED** [review]
Thank you!
